In the following, return type is inferred as (=> a).
Is this the right behavior?
```scala
class LazyValVsFunctionType[a]  {
    val f: a => a = x => {
        lazy val _x: a = throw new java.lang.Error("todo")
        _x // error: type mismatch
/*
[error]  found   : a => => a
[error]  required: a => a
[error]     val f: a => a = x => {
[error]                       ^
[error] one error found
*/
        // _x: a // ok
    }
}
```
(extempore in [r25916](https://codereview.scala-lang.org/fisheye/changelog/scala-svn?cs=25916)) Clean result of function type.

NullaryMethodType was escaping.  Closes #5099, review by moors.
